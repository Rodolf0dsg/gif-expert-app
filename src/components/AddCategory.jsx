import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) => {
        // console.log(target.value);
        setInputValue( target.value );
    }

    const onSubmit = (e) => {
        e.preventDefault();

        if ( inputValue.trim().length <= 1 ) return;

        // console.log( inputValue );
        onNewCategory( inputValue.trim() );
        setInputValue('');
        
    }


    return (
        <form onSubmit={ onSubmit }>
            <input
                type="text" 
                placeholder="Buscar un Gif"
                value={ inputValue }
                onChange={ onInputChange }
            />
        </form>
    )
}

AddCategory.propTypes= {
    onNewCategory: PropTypes.func.isRequired,
}