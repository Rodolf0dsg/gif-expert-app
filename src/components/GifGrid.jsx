import { GifItem } from "./GifItem";

import { useFetchGifs } from "../hooks/useFetchGifs";
// import { getGifs } from "../helpers/getGifs";


export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs( category );
  
    return (
        <>
            <h3>{ category }</h3>

            {
                isLoading && ( <h2>Cargando...</h2> )
            }

            <div className="card-grid">
                {
                    images.map( ( img ) => ( 
                        <GifItem 
                            key={ img.id }
                            { ...img } //se envian todas las props
                        />
                    ))
                }
            </div>
        </>
    )
};

