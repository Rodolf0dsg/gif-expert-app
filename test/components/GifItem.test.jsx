import { GifItem } from "../../src/components/GifItem"
import { render, screen } from "@testing-library/react";

describe('Pruebas en GifItem', () => {

    const title = 'Gif';
    const url = 'https://random-gif.com/random.jpg'

    test('Evaluando el snapshot', () => {

        const { container } = render(<GifItem title={ title } url={ url }  />)

        expect( container ).toMatchSnapshot();

    });

    test('Imagen debe mostrar url y titulo indicado', () => {

        render(<GifItem title={ title } url={ url }  />);
        // screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( title );

        const { src, alt }  = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );

    });

    test('Debe de mostrar el titulo del componente', () => {

        render(<GifItem title={ title } url={ url }  />);
        expect( screen.getByText( title ) ).toBeTruthy();

    });

});