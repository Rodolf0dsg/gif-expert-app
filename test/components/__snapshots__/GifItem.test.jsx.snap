// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Evaluando el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Gif"
      src="https://random-gif.com/random.jpg"
    />
    <p>
      Gif
    </p>
  </div>
</div>
`;
